tests/test_setup.py

```diff
@@ -1,10 +1,22 @@
-from unittest.mock import Mock, patch
```

Comment:
It's a bit confusing for the new `startup` function to be tested in a file called `test_setup`, when `setup` is deprecated. Should we rename the file to `test_startup.py` and rename the previous `test_startup.py` to `test_sqla_startup.py`.  Alternatively we should merge both, but we should make it clear when were testing the wrapper vs the specific implementation.

Reply:
right. I have renamed `test_startup.py` to `test_sqla_startup.py` and split `test_setup.py` into 3 files.